experimental/material_3_demo/lib/main.dart

```diff
@@ -0,0 +1,724 @@
+          useMaterial3: onMaterial3,
+          brightness: Brightness.dark);
+    });
+  }
+
+  void handleColorSelect(value) {
+    setState(() {
+      colorSelected = value;
+      colorSeed = colorOptions[value];
+      lightTheme =
+          ThemeData(colorSchemeSeed: colorSeed, useMaterial3: onMaterial3);
+      darkTheme = ThemeData(
+          colorSchemeSeed: colorSeed,
+          useMaterial3: onMaterial3,
+          brightness: Brightness.dark);
+    });
+  }
+
+  Widget handleScreenChange(screenIndex, showNavBarExample) {
+    switch (screenIndex) {
+      case 0:
+        return ComponentScreen(showNavBottomBar: showNavBarExample);
+      case 1:
+        return const ColorPalettesScreen();
+      case 2:
+        return const TypographyScreen();
+      case 3:
+        return const SurfaceTonesScreen();
+      default:
+        return ComponentScreen(showNavBottomBar: showNavBarExample);
+    }
+  }
+
+  PreferredSizeWidget handleAppBar() {
+    return AppBar(
+      title: onMaterial3 ? const Text("Material 3") : const Text("Material 2"),
+      actions: [
+        IconButton(
+          icon: onLightMode
+              ? const Icon(Icons.wb_sunny_outlined)
+              : const Icon(Icons.wb_sunny),
+          onPressed: handleModeChange,
+        ),
+        IconButton(
+            onPressed: handleVersionChange,
+            icon: onMaterial3
+                ? const Icon(Icons.filter_3)
+                : const Icon(Icons.filter_2)),
+        PopupMenuButton(
+          icon: const Icon(Icons.more_vert),
+          shape:
+              RoundedRectangleBorder(borderRadius: BorderRadius.circular(10)),
+          itemBuilder: (context) {
+            return List.generate(colorOptions.length, (index) {
+              return PopupMenuItem(
+                  value: index,
+                  child: Wrap(
+                    children: [
+                      Padding(
+                        padding: const EdgeInsets.only(left: 10),
+                        child: Icon(
+                          index == colorSelected
+                              ? Icons.color_lens
+                              : Icons.color_lens_outlined,
+                          color: colorOptions[index],
+                        ),
+                      ),
+                      Padding(
+                          padding: const EdgeInsets.only(left: 20),
+                          child: Text(colorText[index]))
+                    ],
+                  ));
+            });
+          },
+          onSelected: handleColorSelect,
+        ),
+      ],
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      debugShowCheckedModeBanner: false,
+      title: 'Material 3',
+      themeMode: onLightMode ? ThemeMode.light : ThemeMode.dark,
+      theme: onLightMode ? lightTheme : darkTheme,
+      home: LayoutBuilder(builder: (context, constraints) {
+        if (constraints.maxWidth < 450) {
```

Comment:
Again, might be good to have the 450 be a named constant to make it clear what is used for here.